lib/linter.js

```diff
@@ -1000,16 +1001,27 @@ module.exports = class Linter {
          * automatically be informed that this type of node has been found
          * and react accordingly.
          */
-        traverser.traverse(sourceCode.ast, {
-            enter(node, parent) {
-                node.parent = parent;
-                eventGenerator.enterNode(node);
-            },
-            leave(node) {
-                eventGenerator.leaveNode(node);
-            },
-            visitorKeys: sourceCode.visitorKeys
-        });
+        try {
+            traverser.traverse(sourceCode.ast, {
+                enter(node, parent) {
+                    node.parent = parent;
+                    eventGenerator.enterNode(node);
+                },
+                leave(node) {
+                    eventGenerator.leaveNode(node);
+                },
+                visitorKeys: sourceCode.visitorKeys
+            });
+        } catch (err) {
+            this._logger("An error occurred while traversing");
+            if (this._context) {
```

Comment:
Was this a typo? I don't think `this._context` exists on `Linter`.

The filename, parser options, etc. should be in scope here already, so it's probably not necessary to pull them from `context`.

Reply:
fixed, cc @not-an-aardvark